java/java.hints/src/org/netbeans/modules/java/hints/errors/EnablePreviewFeature.java

```diff
@@ -0,0 +1,415 @@
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints.errors;
+
+import com.sun.source.util.TreePath;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.project.FileOwnerQuery;
+import org.netbeans.modules.java.hints.spi.ErrorRule;
+import org.netbeans.spi.editor.hints.ChangeInfo;
+import org.netbeans.spi.editor.hints.Fix;
+import org.netbeans.spi.project.support.ant.AntProjectHelper;
+import org.netbeans.spi.project.support.ant.EditableProperties;
+import org.openide.filesystems.FileObject;
+import org.openide.util.NbBundle;
+import org.openide.util.Parameters;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectManager;
+import org.netbeans.modules.maven.api.customizer.ModelHandle2;
+import org.netbeans.modules.maven.execute.model.NetbeansActionMapping;
+import org.netbeans.modules.maven.model.ModelOperation;
+import org.netbeans.modules.maven.model.pom.Build;
+import org.netbeans.modules.maven.model.pom.Configuration;
+import org.netbeans.modules.maven.model.pom.POMComponentFactory;
+import org.netbeans.modules.maven.model.pom.POMExtensibilityElement;
+import org.netbeans.modules.maven.model.pom.POMModel;
+import org.netbeans.modules.maven.model.pom.POMQName;
+import org.netbeans.modules.maven.model.pom.Plugin;
+import org.netbeans.spi.project.ProjectConfiguration;
+import org.netbeans.spi.project.ProjectConfigurationProvider;
+
+import org.openide.filesystems.FileSystem;
+import org.openide.util.Mutex;
+import org.openide.util.MutexException;
+
+/**
+ * Handle error rule "compiler.err.preview.feature.disabled.plural" and provide
+ * the fix for Maven and Ant type project.
+ *
+ * @author arusinha
+ */
+public class EnablePreviewFeature implements ErrorRule<Void> {
+
+    private static final Set<String> ERROR_CODES = new HashSet<String>(Arrays.asList(
+            "compiler.err.preview.feature.disabled.plural")); // NOI18N
+    private static final String ENABLE_PREVIEW_FLAG = "--enable-preview";   // NOI18N
+    private static final String JAVAC_COMPILER_ARGS = "javac.compilerargs"; // NOI18N
+    private static final String RUN_JVMARGS = "run.jvmargs"; // NOI18N
+
+    @Override
+    public Set<String> getCodes() {
+        float jvmVersion = Float.parseFloat(System.getProperty("java.specification.version")); // NOI18N
```

Comment:
Not clear what is the reason to check the java spec. version here? If the error is reported, then we can suggest adding the option, and ignore it othewise?

Reply:
if you have nb-javac plugin installed, error will show even if you are running in jdk-8

Reply:
Yes, but why the fix shouldn't be shown in that case? The project should have JDK 12+ configured as its platform, so the option will work? (Maybe check if CompilationInfo.getSourceVersion() == SourceVersion.lastest(), to avoid some clashes between --enable-preview and -source, but I don't see how the IDE's runtime platform could impact the decisions here.)